### PR TITLE
fix(ckb-indexer): wrong case handling when collect tx

### PR DESCRIPTION
**Pull request by homura**
# Description

`TransactionCollector` forgot to convert the response to the camel case when collect tx

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Added unit test

### CODE REVIEW

File: packages/ckb-indexer/src/services.ts [hunk 1]
```diff
@@ -86,6 +96,38 @@ const requestBatch = async (rpcUrl: string, data: unknown): Promise<any> => {
...
+  const requestBody = txHashes.map((txHash, index) => {
+    return {
+      id: index,
+      jsonrpc: "2.0",
+      method: "get_transaction",
+      params: [txHash],
+    };
+  });
+
+  type RpcTransactionWithStatus = Parameters<
```
Comment (zhangyouxin) [inline packages/ckb-indexer/src/services.ts]: There is a `TransactionWithStatus` type in `@ckb-lumos/rpc/lib/types/rpc`:
```ts
import type { RPC as RPCTypes } from '@ckb-lumos/rpc/lib/types/rpc'
type RpcTransactionWithStatus = RPCTypes.TransactionWithStatus
```

File: packages/ckb-indexer/src/indexer.ts [hunk 2]
```diff
@@ -271,31 +270,24 @@ export class CkbIndexer implements CellProvider, TerminableCellFetcher {
...
-              const cellIndex = tx.inputs[index].previousOutput.index;
-              const outputCell =
-                item.result.transaction.outputs[parseInt(cellIndex)];
-              const outputData =
-                item.result.transaction.outputsData[parseInt(cellIndex)];
-              return { output: outputCell, outputData } as OutputToVerify;
+        const transactionResponse: OutputToVerify[] =
+          await requestBatchTransactionWithStatus(this.uri, inputTxHashes).then(
+            (response) => {
+              return response.map((txWithStatus, index: number) => {
```
Comment (zhangyouxin) [inline packages/ckb-indexer/src/indexer.ts]:
```suggestion
              return response.map((txWithStatus, index) => {
```

File: packages/ckb-indexer/tests/transaction_collector.unit.test.ts [hunk 3]
```diff
@@ -76,70 +79,234 @@ test.serial(
...
+        if (!txs[hash]) {
+          throw new Error("Transaction not found" + hash);
+        }
+        return txs[hash];
+      });
+    });
+
+    const transactionCollector = new TransactionCollector(
+      scopedIndexer,
+      queryOption,
```
Comment (zhangyouxin) [inline packages/ckb-indexer/tests/transaction_collector.unit.test.ts:172]: we should test `transactionCollector` with different queryOptions

Comment (homura) [reply packages/ckb-indexer/tests/transaction_collector.unit.test.ts:172]: There are some different `queryOptions` in the following test cases

**Comment by zhangyouxin** [comment]
better make it pass codecov check

**Comment by homura** [comment]
> better make it pass codecov check

Most of the uncovered lines are in the `ckb-indexer.ts`, I think we can improve this part of the unit test in a new issue

File: packages/testkit/src/mock-ckb-jsonrpc.ts [hunk 4]
```diff
@@ -90,7 +91,7 @@ export function createCKBMockRPC(options: Options): Express {
     }
     return {
       transaction: result,
-      txStatus: { status: "padding", blockHash: blockHash },
+      tx_status: { status: "padding", block_hash: blockHash },
```
Comment (zhangyouxin) [inline packages/testkit/src/mock-ckb-jsonrpc.ts]:
```suggestion
      tx_status: { status: "pending", block_hash: blockHash },
```

**Review by zhangyouxin — APPROVED** [review]

**Pull request merged**